diff --git a/sdl2-sys/SDL_ttf/CMakeLists.txt b/sdl2-sys/SDL_ttf/CMakeLists.txt
index e52d0481bb69cf50d8003f7de2f43ac22cf12ae7..c5a5a49e0cd3d10286e4d4e23bf20cdf626abd9a 100644
--- a/sdl2-sys/SDL_ttf/CMakeLists.txt
+++ b/sdl2-sys/SDL_ttf/CMakeLists.txt
@@ -54,11 +54,13 @@ include(GNUInstallDirs)
 include(CheckSymbolExists)
 
 option(CMAKE_POSITION_INDEPENDENT_CODE "Build static libraries with -fPIC" ON)
-option(BUILD_SHARED_LIBS "Build the library as a shared library" ON)
+option(BUILD_SHARED_LIBS "Build the library as a shared library" ${SDL_SHARED})
 
 option(SDL2TTF_SAMPLES "Build the SDL2_ttf sample program(s)" ${SDL2TTF_ROOTPROJECT})
 cmake_dependent_option(SDL2TTF_INSTALL "Enable SDL2_ttf install target" ${SDL2TTF_ROOTPROJECT} "${sdl2ttf_install_enableable}" OFF)
-option(SDL2TTF_VENDORED "Use vendored third-party libraries" ${vendored_default})
+if(NOT DEFINED SDL2TTF_VENDORED)
+  option(SDL2TTF_VENDORED "Use vendored third-party libraries" ${vendored_default})
+endif()
 
 # For style consistency, create a SDL2TTF_FREETYPE CMake variable. This variable is NOT configurable.
 set(SDL2TTF_FREETYPE ON)
